lib/awful/widget/layoutlist.lua

```diff
@@ -0,0 +1,442 @@
+--    )
+--
+-- @property buttons
+-- @param table
+
+--- The default foreground (text) color.
+-- @beautiful beautiful.layoutlist_fg_normal
+-- @tparam[opt=nil] string|pattern fg_normal
+-- @see gears.color
+
+--- The default background color.
+-- @beautiful beautiful.layoutlist_bg_normal
+-- @tparam[opt=nil] string|pattern bg_normal
+-- @see gears.color
+
+--- The selected layout foreground (text) color.
+-- @beautiful beautiful.layoutlist_fg_selected
+-- @tparam[opt=nil] string|pattern fg_selected
+-- @see gears.color
+
+--- The selected layout background color.
+-- @beautiful beautiful.layoutlist_bg_selected
+-- @tparam[opt=nil] string|pattern bg_selected
+-- @see gears.color
+
+--- Disable the layout icons (only show the name label).
+-- @beautiful beautiful.layoutlist_disable_icon
+-- @tparam[opt=false] boolean layoutlist_disable_icon
+
+--- Disable the layout name label (only show the icon).
+-- @beautiful beautiful.layoutlist_disable_name
+-- @tparam[opt=false] boolean layoutlist_disable_name
+
+--- The layoutlist font.
+-- @beautiful beautiful.layoutlist_font
+-- @tparam[opt=nil] string font
+
+--- The selected layout alignment.
+-- @beautiful beautiful.layoutlist_align
+-- @tparam[opt=left] string align *left*, *right* or *center*
+
+--- The selected layout title alignment.
+-- @beautiful beautiful.layoutlist_font_selected
+-- @tparam[opt=nil] string font_selected
+
+--- The space between the layouts.
+-- @beautiful beautiful.layoutlist_spacing
+-- @tparam[opt=0] number spacing The spacing between tasks.
+
+--- The default layoutlist elements shape.
+-- @beautiful beautiful.layoutlist_shape
+-- @tparam[opt=nil] gears.shape shape
+
+--- The default layoutlist elements border width.
+-- @beautiful beautiful.layoutlist_shape_border_width
+-- @tparam[opt=0] number shape_border_width
+
+--- The default layoutlist elements border color.
+-- @beautiful beautiful.layoutlist_shape_border_color
+-- @tparam[opt=nil] string|color shape_border_color
+-- @see gears.color
+
+--- The selected layout shape.
+-- @beautiful beautiful.layoutlist_shape_selected
+-- @tparam[opt=nil] gears.shape shape_selected
+
+--- The selected layout border width.
+-- @beautiful beautiful.layoutlist_shape_border_width_selected
+-- @tparam[opt=0] number shape_border_width_selected
+
+--- The selected layout border color.
+-- @beautiful beautiful.layoutlist_shape_border_color_selected
+-- @tparam[opt=nil] string|color shape_border_color_selected
+-- @see gears.color
+
+--- The currenly displayed layouts.
+-- @property layouts
+-- @param table
+
+--- The currently selected layout.
+-- @property current_layout
+-- @param layout
+
+function layoutlist:get_layouts()
+    local f = self.source or self._private.source or module.source.for_screen
+
+    return f(self.screen)
+end
+
+function layoutlist:get_current_layout()
+    -- Eventually the entire lookup could be removed. All it does is allowing
+    -- `nil` to be returned when there is no selection.
```

Comment:
Would we want an explicit `return nil` at the end of the function?

Reply:
Done